File: 04_quicksort/02_recursive_sum.js
//  Sums values in the array using recursion

function sumRecursive(array) {
  if (array.length == 1) { // base case
    return array[0];
  } else {
    return array[0] + sumRecursive(array.slice(1));
  }
}
console.log(sumRecursive([1, 2, 3, 4])); // 10

File: 04_quicksort/02_recursive_sum.rb
# Sums values in the array using recursion

def sumRecursive(array)
  return 0 if array.empty? # base case
  array[0] + sumRecursive(array.slice(1..-1))
end

p sumRecursive([1, 2, 3, 4]) # 10
